src/test/java/com/epam/tc/hw2/ex1/Ex1.java

```diff
@@ -0,0 +1,174 @@
+                .as("To be flexible...text differs").isEqualTo("To be flexible and\ncustomizable");
+        softly.assertThat(iconTexts.get(2).getText())
+                .as("To be multiplatform").isEqualTo("To be multiplatform");
+        softly.assertThat(iconTexts.get(3).getText())
+                .as("Already have good base...text differs")
+                .isEqualTo("Already have good base\n"
+                        + "(about 20 internal and\n"
+                        + "some external projects),\nwish to get more…");
+
+        //8. Assert that there is the iframe with “Frame Button” exist
+        int size = driver.findElements(By.tagName("iframe")).size();
+        System.out.println(size);
+        int buttonInCurrentFrame = 0; //count buttons “Frame Button” in current frame
+        int totalButtonsInFrames = 0; //counter of frames with button “Frame Button”
+        for (int i = 0; i < size; i++) {
+
+            //9. Switch to the iframe and check that there is “Frame Button” in the iframe
+            driver.switchTo().frame(i);
+            buttonInCurrentFrame = driver.findElements(By.cssSelector("input#frame-button")).size();
+
+            if (buttonInCurrentFrame > 0) {
+                totalButtonsInFrames = 0 + buttonInCurrentFrame;
+                WebElement frameButton = driver
+                        .findElement(By.cssSelector("input#frame-button"));
+                softly.assertThat(frameButton.getAttribute("value"))
+                        .as("Frame Button text differs")
+                        .isEqualTo("Frame Button");
+            }
+
+            System.out.println(buttonInCurrentFrame);
+
+            //10. Switch to original window back
+            driver.switchTo().defaultContent();
+        }
+        softly.assertThat(totalButtonsInFrames).as("Button not found").isNotZero();
+
+        //11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        List<WebElement> sidebarItems = driver.findElements(By.cssSelector(".sidebar-menu > li"));
+        softly.assertThat(sidebarItems.size())
+                .as("Number of menu items " + sidebarItems.size() + " not 5").isEqualTo(5);
+        softly.assertThat(sidebarItems.get(0)
+                .getText()).as("Home menu text differs").isEqualTo("Home");
+        softly.assertThat(sidebarItems.get(1)
+                .getText()).as("Contact form text differs").isEqualTo("Contact form");
+        softly.assertThat(sidebarItems.get(2)
+                .getText()).as("Service text differs").isEqualTo("Service");
+        softly.assertThat(sidebarItems.get(3)
+                .getText()).as("Metals & Colors text differs").isEqualTo("Metals & Colors");
+        softly.assertThat(sidebarItems.get(4)
+                .getText()).as("Elements text packs").isEqualTo("Elements packs");
+
+        softly.assertAll();
+
+        //12. Close Browser
+        driver.quit();
+
```

Comment:
If test fail what will happen to the browser and webdriver? Maybe you can move tear down step somewhere else?

Reply:
Moved to postconditions for this homevork and to parent class in the next one.